config.go

```diff
@@ -258,22 +259,22 @@ func LoadConfig(filePath *string, configStruct *TykPumpConfiguration) {
 	}
 	configStruct.Pumps = toUpperMap
 
-	shouldOmit, omitEnvExist := os.LookupEnv(ENV_PREVIX + "_OMITCONFIGFILE")
-	if configStruct.OmitConfigFile || (omitEnvExist && strings.ToLower(shouldOmit) == "true") {
-		*configStruct = TykPumpConfiguration{}
-	}
-
 	overrideErr := envconfig.Process(ENV_PREVIX, configStruct)
 	if overrideErr != nil {
 		log.Error("Failed to process environment variables after file load: ", overrideErr)
 	}
 
 	errLoadEnvPumps := configStruct.LoadPumpsByEnv()
 	if errLoadEnvPumps != nil {
-		log.Fatal("error loading pumps env vars:", err)
+		log.Fatal("error loading pumps env vars:", errLoadEnvPumps)
 	}
 }
 
+func (cfg *TykPumpConfiguration) shouldOmitConfigFile() bool {
+	shouldOmit, omitEnvExist := os.LookupEnv(ENV_PREVIX + "_OMITCONFIGFILE")
+	return cfg.OmitConfigFile || (omitEnvExist && strings.EqualFold(shouldOmit, "true"))
```

Comment:
We can remove the `cfg.OmitConfigFile` check here - if we are checking this before loading the configuration it will always be false.